packages/builtin/array.pony

```diff
@@ -63,11 +63,11 @@ class Array[A] is Seq[A]
     """
     _ptr._offset(from_offset)._copy_to(ptr._offset(to_offset), copy_len)
 
-  fun cpointer(): Pointer[A] tag =>
+  fun cpointer(offset: USize = 0): Pointer[A] tag =>
     """
     Return the underlying C-style pointer.
     """
-    _ptr
+    _ptr._offset(offset)
```

Comment:
The main concern would be that there is no bounds checking prior to using `Pointer._offset`. I'm not sure if the intrinsic for `Pointer._offset` handles that or not.

Let me know if I should add bounds checking.